evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -26,10 +32,126 @@ def __init__(self, parameters=None, component_obj=None, random_seed=0, **kwargs)
+
+        Args:
+            X (pandas.DataFrame): The feature data of the time series problem.
+            y (pandas.Series): The target data of a time series problem.
+            method (str): Either "autocorrelation" or "partial-autocorrelation".  The method by which to determine the
+                first period of the seasonal part of the target signal.  "partial-autocorrelation" should currently not
+                be used.  Defaults to "autocorrelation".
+
+        Returns:
+            (list[int]): The integer numbers of entries in time series data over which the seasonal part of the target data
+                repeats.  If the time series data is in days, then this is the number of days that it takes the target's
+                seasonal signal to repeat. Note: the target data can contain multiple seasonal signals.  This function
+                will only return the first, and thus, shortest period.  E.g. if the target has both weekly and yearly
+                seasonality, the function will only return "7" and not return "365".  If no period is detected, returns [None].
+
+        """
+
+        def _get_rel_max_from_acf(y):
+            acf = sm.tsa.acf(y, nlags=np.maximum(400, len(y)))
+            filter_acf = [acf[i] if (acf[i] > 0) else 0 for i in range(len(acf))]
+            rel_max = argrelextrema(
+                np.array(filter_acf),
+                np.greater,
+                order=5,  # considers 5 points on either side to determine rel max
+            )[0]
+            max_acfs = [acf[i] for i in rel_max]
+            if len(max_acfs) > 0:
+                rel_max = np.array([filter_acf.index(max(max_acfs))])
+            else:
+                rel_max = []
+            return rel_max
+
+        def _get_rel_max_from_pacf(y):
+            pacf = sm.tsa.pacf(y)
+            return argrelextrema(pacf, np.greater)[0]
+
+        def _detrend_on_fly(X, y):
+            self.fit(X, y)
+            res = self.get_trend_dataframe(X, y)
+            y_time_index = self._set_time_index(X, y)
+            y_detrended = y_time_index - res[0]["trend"]
+            return y_detrended
+
+        if method == "autocorrelation":
+            _get_rel_max = _get_rel_max_from_acf
+        elif method == "partial-autocorrelation":
+            self.logger.warning(
+                "Partial autocorrelation should not be used yet.  Resulting period might not be accurate.",
+            )
+            _get_rel_max = _get_rel_max_from_pacf
+
+        # Make the data more stationary by detrending
+        y_detrended = _detrend_on_fly(X, y)
+        relative_maxima = _get_rel_max(y_detrended)
+        self.logger.warning(
+            f"Decomposer discovered {len(relative_maxima)} possible periods.",
+        )
+
+        if len(relative_maxima) == 0:
+            self.logger.warning("No periodic signal could be detected in target data.")
+            relative_maxima = [None]
+        return relative_maxima[0]
```

Comment:
Clarification: is the first element of this list guaranteed to be the "most likely" period if there are multiple possible periods?

Reply:
Nope!  This PR is to come up with our best, first guess.  I've filed half a dozen improvement issues, but here's the most relevant one associated with multiple frequencies present in the data : https://alteryx.atlassian.net/browse/OSSEVALML-395?atlOrigin=eyJpIjoiZjJlMjdlZjE5MzYxNDhlMGE4Njg4ZjhjYTc2MjY5MDgiLCJwIjoiaiJ9